### PR TITLE
Allow users to override our default behavior of rewrite rule-ids

### LINKED ISSUE #290: The returntocorp/semgrep-agent does not include the “--no-rewrite-rule-ids” option
Currently we do not have the option to ignore findings while using the "semgrep-agent" in our CI pipeline.  Given the Rule IDs are being rewritten with a random CI directory name we are unable to add the Rule ID to a “nosemgrep: rule-id” inline comment.

Example renaming of a Rule ID:
```
tmp.tmp.ggDeOk.semgrep-rules.typescript.react.security.audit.react-missing-noopener.react-missing-noopener
```

Error message from semgrep-agent:
```
Error: no such option: --no-rewrite-rule-ids
```

### PR DESCRIPTION
**Pull request by chmccreery**
Allow users to override our default behavior of rewriting rule-ids when their rules are stored in a .semgrep/ directory, the same way they would be able to with semgrep CLI.

Fixes #290

### CODE REVIEW

File: src/semgrep_agent/main.py
Comment (brendongo) [inline src/semgrep_agent/main.py:333]: Should we allow this to be set to false if sapp.is_configured is true? i.e. should this be an or?

Comment (chmccreery) [reply src/semgrep_agent/main.py:333]: I don't think so - if they are using semgrep cloud and try to rewrite rule ids, semgrepignores will no longer work and rule names will be pre-pended with tmp.waf93 or something I think. Since all their rules will be in one file, there is strictly no benefit to having no-rewrite-rule-ids set.

Comment (chmccreery) [reply src/semgrep_agent/main.py:333]: Perhaps that means I should change the flag to not let them set --rewrite-rule-ids to true ever, but I didn't like how many double-negatives that put into the logic. It made it hard to reason about.

Comment (brendongo) [reply src/semgrep_agent/main.py:333]: So if we always want rules from semgrep-cloud to rewrite rule_ids shouldn't this be `rewrite_rule_ids or not sapp.is_configured`?

Comment (chmccreery) [reply src/semgrep_agent/main.py:333]: We do not always want rules from semgrep-cloud to rewrite rule_ids. We actually _never_ want rules from semgrep-cloud to rewrite rule_ids because rules are downloaded and saved to a flat config file - there is no meaningful path to prefix the rules with. The logic is:

if `sapp.is_configured` is true, then `not sapp.is_configured` is false, so anything anded with `not sapp.is_configured` is also false, so rule ids will never be rewritten ✅

Comment (chmccreery) [reply src/semgrep_agent/main.py:333]: Meanwhile, if `sapp.is_configured` is false, then `not sapp.is_configured` is true, so then we follow whatever is specified by the commandline flag ✅

Comment (brendongo) [reply src/semgrep_agent/main.py:333]: Ah makes sense. I got confused with all the nots and rewrite-not-rewrite :D

**Comment by brendongo** [comment]
This looks good. i'm wondering if we should revisit https://github.com/returntocorp/semgrep-action/pull/169 since once we add this flag it will be involved to remove it later.

File: src/semgrep_agent/main.py
Comment (chmccreery) [reply src/semgrep_agent/main.py:333]: Yeah, it is confusing 🙁 but I couldn't figure out how to write this in a less-confusing way given how semgrep already does it

**Comment by chmccreery** [comment]
> This looks good. i'm wondering if we should revisit #169 since once we add this flag it will be involved to remove it later.

Yeah, good to keep #169 in mind. I'll defer to @dlukeomalley on whether this bug fix is still high-priority or whether we should pause and not merge this until/unless #169 gets resolved 🤷

**Review by brendongo — APPROVED** [review]
Approving to unblock since it looks like https://github.com/returntocorp/semgrep-action/pull/169 needs more work still

**Pull request merged**
